content/02.body.md

```diff
@@ -13,6 +13,22 @@
 
 ### XSwap algorithm
 
+Hanhijärvi, et al. presented XSwap [@doi:10.1137/1.9781611972795.67], an algorithm for the randomization ("permutation") of unweighted networks (Figure {@fig:algo}A).
+The algorithm picks two existing edges at random, and if the edges constitute a valid swap, exchanges the targets between the edges (Figure {@fig:xswap}).
+
+![Graphical representation of the XSwap algorithm applied to two edges.
+The algorithm preserves both the source- and target-degree for all nodes.](images/xswap_figure.png){#fig:xswap width="50%"}
+
+To allow greater flexibility, we modified the algorithm by adding two parameters, "`allow_self_loops`", and "`allow_antiparallel`" that allow a greater variety of network types to be permuted (Figure {@fig:algo}B).
```

Comment:
Are these two options sufficient to address the different network types: directed, undirected, bipartite? It seems like we should mention our handling of these

Reply:
I prefer to go with @cgreene s suggestion to move these to the hetio/xswap repository documentation. I link it in https://github.com/greenelab/xswap-manuscript/pull/16/commits/2179b0a582ae5b40315989838fb3fd58abfc054c with 

"We provide documentation for parameter choices depending on the type of network being permuted in the GitHub repository (https://github.com/hetio/xswap)."

Link to line is https://github.com/greenelab/xswap-manuscript/pull/16/commits/2179b0a582ae5b40315989838fb3fd58abfc054c#diff-e4cf93b0b53712cbe69b13f71ee7d779R25